File: Contador/Cnt.py
def contar_texto(texto):
    caracteres = 0
    palabras = 0
    oraciones = 0
    dentro_palabra = False

    for c in texto:
        # Contar caracteres (sin espacios, tabulaciones ni saltos de línea)
        if c not in [' ', '\t', '\n']:
            caracteres += 1

        # Detectar palabras
        if c not in [' ', '\t', '\n']:
            if not dentro_palabra:
                palabras += 1
                dentro_palabra = True
        else:
            dentro_palabra = False

        # Detectar oraciones
        if c in ['.', '?', '!']:
            oraciones += 1

    return caracteres, palabras, oraciones


# Ejemplo de uso
texto = input("Ingrese el texto: ")
caracteres, palabras, oraciones = contar_texto(texto)

print("Numero de caracteres (sin espacios):", caracteres)
print("Numero de palabras:", palabras)
print("Numero de oraciones:", oraciones)

File: Contador/Cnt.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    string texto;
    cout << "Ingrese el texto: ";
    getline(cin, texto);

    int caracteres = 0;
    int palabras = 0;
    int oraciones = 0;
    bool dentroPalabra = false;

    for (char c : texto) {
        // Contar caracteres (sin espacios, tabulaciones ni saltos de línea)
        if (c != ' ' && c != '\t' && c != '\n') {
            caracteres++;
        }

        // Detectar palabras
        if (c != ' ' && c != '\t' && c != '\n') {
            if (!dentroPalabra) {
                palabras++;
                dentroPalabra = true;
            }
        } else {
            dentroPalabra = false;
        }

        // Detectar oraciones (terminan en ., ?, !)
        if (c == '.' || c == '?' || c == '!') {
            oraciones++;
        }
    }

    cout << "Numero de caracteres (sin espacios): " << caracteres << endl;
    cout << "Numero de palabras: " << palabras << endl;
    cout << "Numero de oraciones: " << oraciones << endl;

    return 0;
}
